lib/Fmt_ast.ml

```diff
@@ -1566,6 +1581,51 @@ and fmt_infix_op_args ?ext c ~parens xexp op_args =
     ~parens_nested:(Ast.parenze_nested_exp xexp)
     (list_fl groups fmt_op_arg_group)
 
+and fmt_infix_op_args_pat ?ext c ~parens xpat op_args =
+  let groups =
+    let not_simple (_, arg) = not (Pat.is_simple arg.ast) in
+    let exists_not_simple args = List.exists args ~f:not_simple in
+    let break (has_cmts, _, _, (_, args1)) (_, _, _, (_, args2)) =
+      has_cmts || exists_not_simple args1 || exists_not_simple args2
+    in
+    match c.conf.break_infix with
+    | `Wrap -> List.group op_args ~break
+    | `Fit_or_vertical -> List.map ~f:(fun x -> [x]) op_args
+  in
+  let is_not_indented {ast= pat; _} =
+    match pat.ppat_desc with
+    | Ppat_open _ -> Source.is_long_ppat_open c.source pat
+    | _ -> false
+  in
+  let fmt_arg _very_last ~first:_ ~last ((_, xarg) as lbl_xarg) =
+    let parens = parenze_pat xarg in
+    fmt_label_arg_pat c ~parens lbl_xarg $ fmt_if (not last) "@ "
+  in
+  let fmt_op_arg_group ~first:first_grp ~last:last_grp args =
+    let indent = if first_grp && parens then -2 else 0 in
+    hovbox indent
+      (list_fl args
+         (fun ~first ~last (_, cmts_before, cmts_after, (op, xargs)) ->
+           let very_first = first_grp && first in
+           let very_last = last_grp && last in
+           cmts_before
+           $ hvbox 0
+               ( op
+               $ ( match xargs with
+                 | (_, e) :: _ when very_last && is_not_indented e ->
+                     fmt "@ "
+                 | _ -> fmt_if (not very_first) " " )
```

Comment:
Do you have an example where `Ppat_open` needs a break after it ? All the tests run fine with
```suggestion
               $ fmt_if (not very_first) " "
```

Reply:
not necessary indeed